From: Tedd Ho-Jeong An <tedd.an@intel.com>
Date: Mon, 4 Oct 2021 10:01:26 -0700
Subject: [PATCH] Bluetooth: hci_vhci: Fix to set the force_wakeup value
References: bsc#1012628
Patch-mainline: 5.16.3
Git-commit: 8b89637dbac2d73d9f3dadf91b4a7dcdb1fc23af

[ Upstream commit 8b89637dbac2d73d9f3dadf91b4a7dcdb1fc23af ]

This patch sets the wakeup state of the vhci driver when the
force_wakeup is updated.

Fixes: 60edfad4fd0b6 ("Bluetooth: hci_vhci: Add force_prevent_wake entry")
Signed-off-by: Tedd Ho-Jeong An <tedd.an@intel.com>
Signed-off-by: Marcel Holtmann <marcel@holtmann.org>
Signed-off-by: Sasha Levin <sashal@kernel.org>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/bluetooth/hci_vhci.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/bluetooth/hci_vhci.c b/drivers/bluetooth/hci_vhci.c
index b45db0db347c..5fd91106e853 100644
--- a/drivers/bluetooth/hci_vhci.c
+++ b/drivers/bluetooth/hci_vhci.c
@@ -176,6 +176,8 @@ static ssize_t force_wakeup_write(struct file *file,
 	if (data->wakeup == enable)
 		return -EALREADY;
 
+	data->wakeup = enable;
+
 	return count;
 }
 
-- 
2.34.1

